src/ThemedWrappers.h

```diff
@@ -0,0 +1,154 @@
+/*!********************************************************************
+
+   Audacity: A Digital Audio Editor
+
+   @file ThemedWrappers.h
+
+   @author Vitaly Sverchinsky
+
+   A set of classes which decorate given base classes with
+   simplified theme support. Note that classes are declared
+   final, which means they could be only used 'in-place'.
+   
+**********************************************************************/
+
+#pragma once
+
+#include <optional>
+
+#include "Internat.h"
+#include "Observer.h"
+#include "Theme.h"
+#include "Prefs.h"
+
+template<class WindowBase>
+class ThemedWindowWrapper final :
+   public WindowBase,
+   public PrefsListener
+{
+   Observer::Subscription mThemeChangeSubscription;
+   std::optional<TranslatableString> mTranslatableLabel;
+   int mForegroundColorIndex { -1 };
+   int mBackgroundColorIndex { -1 };
+public:
+   template <typename... Args>
+   ThemedWindowWrapper(Args&&... args) : WindowBase( std::forward<Args>(args)... )
+   {
+      mThemeChangeSubscription =
+         theTheme.Subscribe(*this, &ThemedWindowWrapper::OnThemeChange);
+   }
+
+   void SetBackgroundColorIndex(int index)
+   {
+      mBackgroundColorIndex = index;
+      if(index != -1)
+         WindowBase::SetBackgroundColour(theTheme.Colour(mBackgroundColorIndex));
+   }
+
+   void SetForegroundColorIndex(int index)
+   {
+      mForegroundColorIndex = index;
+      if(index != -1)
+         WindowBase::SetForegroundColour(theTheme.Colour(mForegroundColorIndex));
+   }
+
+   void SetTranslatableLabel(TranslatableString label)
+   {
+      mTranslatableLabel = std::move(label);
+      WindowBase::SetLabel(mTranslatableLabel->Translation());
+   }
+
+   void UpdatePrefs() override
+   {
+      if(mTranslatableLabel)
+         WindowBase::SetLabel(mTranslatableLabel->Translation());
```

Comment:
Should it also call through to the base class’s UpdatePrefs?

I think yes in general, but it might call through to a pure virtual.  Which might be given a default do nothing implementation.  (Did you know that a pure virtual can have a definition but also remain pure?  Yes it can, so subclasses must define it to be constructible, while qualified calls to the function are also allowed.)